pkg/node/node_address_linux.go

```diff
@@ -28,116 +28,141 @@ import (
 	"github.com/vishvananda/netlink"
 )
 
-// firstGlobalV4Addr returns the first IPv4 global IP of an interface, where
-// the IPs are sorted in ascending order. when intf is defined only IPs
-// belonging to that interface are considered. If preferredIP is present in the
-// IP list it is returned irrespective of the sort order. Passing intf and
-// preferredIP will only return preferredIP if it is in the IPs that belong to
-// intf. In all cases, if intf is not found all interfaces are considered.
-func firstGlobalV4Addr(intf string, preferredIP net.IP) (net.IP, error) {
+var privateIPBlocks []*net.IPNet
+
+func init() {
```

Comment:
You don't need init at all, you can build something lilke this: 

`var privateIPBlocks []*net.IPNet = []*net.IPNet{net.ParseCIDR("10.0.0.0/8"), net.ParseCIDR("172.16.0.0/12")}` 

^^ with all IPS

Reply:
`net.ParseCIDR()` returns multi-value whereas `[]*net.IPNet{}` is single value, so as-is this won't work, so I went with the `init()` function. I presume technically in either case it's a constructor which is called upon init.